src/html2pdf.js

```diff
@@ -32,356 +32,412 @@
+                source = cloneNode(source, opt.html2canvas.javascriptEnabled);
+            } else {
+                throw 'Invalid source - please specify an HTML Element or string.';
+            }
+
+            // Return the parsed input (opt is modified in-place, no need to return).
+            return source;
+        };
+
+        html2pdf.makeContainer = function (source, pageSize) {
+            // Define the CSS styles for the container and its overlay parent.
+            var overlayCSS = {
+                position: 'fixed', overflow: 'hidden', zIndex: 1000,
+                left: 0, right: 0, bottom: 0, top: 0,
+                backgroundColor: 'rgba(0,0,0,0.8)'
+            };
+            var containerCSS = {
+                position: 'absolute', width: pageSize.inner.width + pageSize.unit,
+                left: 0, right: 0, top: 0, height: 'auto', margin: 'auto',
+                backgroundColor: 'white'
+            };
+
+            // Set the overlay to hidden (could be changed in the future to provide a print preview).
+            overlayCSS.opacity = 0;
+
+            // Create and attach the elements.
+            var overlay = createElement('div', {className: 'html2pdf__overlay', style: overlayCSS});
+            var container = createElement('div', {className: 'html2pdf__container', style: containerCSS});
+            container.appendChild(source);
+            overlay.appendChild(container);
+            document.body.appendChild(overlay);
+
+            // Enable page-breaks.
+            var pageBreaks = source.querySelectorAll('.html2pdf__page-break');
+            var pxPageHeight = pageSize.inner.height * pageSize.k / 72 * 96;
+            Array.prototype.forEach.call(pageBreaks, function (el) {
+                el.style.display = 'block';
+                var clientRect = el.getBoundingClientRect();
+                el.style.height = pxPageHeight - (clientRect.top % pxPageHeight) + 'px';
+            }, this);
+
+            // Return the container.
+            return container;
+        };
+
+        html2pdf.makePDF = function (canvas, pageSize, opt) {
+            // Calculate the number of pages.
+            var ctx = canvas.getContext('2d');
+            var pxFullHeight = canvas.height;
+            var pxPageHeight = Math.floor(canvas.width * pageSize.inner.ratio);
+            var nPages = Math.ceil(pxFullHeight / pxPageHeight);
+            var pageModulus = pxFullHeight % pxPageHeight;
+
+            // Create a one-page canvas to split up the full image.
+            var pageCanvas = document.createElement('canvas');
+            var pageCtx = pageCanvas.getContext('2d');
+            var pageHeight = pageSize.inner.height;
+            pageCanvas.width = canvas.width;
+            pageCanvas.height = pxPageHeight;
+
+            // Initialize the PDF.
+            var pdf = new jsPDF(opt.jsPDF);
+
+            /**
+             * added support for page split only on white "rows".
```

Comment:
This should probably be an "option" (configurable in html2pdf settings) rather than hard-coded to always run. That way background color, max pixels, etc. could be configured as well.

Reply:
great idea - also maybe add the option to decide what level is for "white" (the 150 value i put in)